authentication/models/user.py

```diff
@@ -37,48 +39,37 @@ def _create_user(self, id, first_name, last_name, email, password, birthday, pro
 
 
 class User(AbstractBaseUser):
-    id = models.CharField(primary_key=True, max_length=30, verbose_name="User id")
-    first_name = models.CharField(max_length=50, verbose_name="User first name")
-    last_name = models.CharField(max_length=50, verbose_name="User last name")
-    promo = models.IntegerField(verbose_name="Promotion")
-
-    email = models.EmailField(
-        verbose_name='email address',
-        max_length=160,
-        unique=True,
-    )
+    id = models.CharField(primary_key=True, max_length=30)
 
+    first_name = models.CharField(max_length=50)
+    last_name = models.CharField(max_length=50)
     nickname = models.CharField(max_length=128, blank=True, default="")
-    birthday = models.DateField(null=True, verbose_name="date de naissance")
-
-    # Contact the person
-    phone = models.CharField(max_length=15, blank=True, verbose_name="numéro de téléphone")
+    birthday = models.DateField(null=True)
+    email = models.EmailField(max_length=160, unique=True)
+    promo = models.IntegerField(verbose_name="Promotion",
+                                help_text="The last two digits of the User's entrance year at school.")
 
-    room = models.CharField(max_length=128, blank=True, verbose_name="numéro de chambre")  # null if the person is PAM
+    phone = models.CharField(max_length=15, blank=True)
+    room = models.CharField(max_length=128, blank=True, help_text="Blank if the User is PAM.")
     address = models.CharField(max_length=512, blank=True,
-                               help_text="adresse en dehors de la Meuh")  # null if the person is not PAM
-    city_of_origin = models.CharField(max_length=128, blank=True, help_text="ville d'origine")
+                               help_text="Address outside the Meuh. Blank if the User is not PAM.")
+    city_of_origin = models.CharField(max_length=128, blank=True)
 
-    # Cursus
+    # Education.
     option = models.CharField(max_length=128, blank=True)
     is_ast = models.BooleanField(default=False)
     is_isupfere = models.BooleanField(default=False)
     is_in_gapyear = models.BooleanField(default=False)
```

Comment:
Maybe change the name of this variable to keep it even after the perso has come back from gapyear ?

Reply:
Problem is, if we change it to “did_a_gapyear”, we cannot know if he or she is currently in a gapyear or not (because of people who repeat years).

Reply:
This raises the more general issue of people who repeated a year. They might be excluded from votes for instance

Reply:
I would be in favour of creating something like a new Model AcademicYear with three fields:
* the user;
* the beginning year of the academic year (such as 2017 for 2017-2018);
* the academic year (1A, 2A, Césure…).

It would also help categorizing the AST as people who must not have access to the secret…

Reply:
BTW, I think we should remove the field is_ast, and replace it by student_type (or something like this), with options like 'AST', 'EV', 'ISUPFERE', 'IC'…

Reply:
The `AcademicYear` could be directly integrated to `User`. Explicitly stating the academic year sounds good, although I'm not sure to get the point if having the beginning of the year

Reply:
OK, so a `list<tuple<int, str>>` such as `[(2017, '1A'), (2018, '1A'), (2019, '2A'), (2020, 'GAP_YEAR'), (2021, '3A')]`?
But how do we enforce, in Django, that the second member of the `tuple` is always something in `1A`, `2A`, …?

By the beginning year I mean, instead of storing 2017-2018 which is a string, only store 2017, which is a nice integer.

Reply:
Or maybe you meant why do we need to store the academic year (2017-2018 or 2017 or whatever) as we have the “promotion”?
Well, maybe some people will leave the school or whatever. And your promotion may change. Explicit is better than implicit, as they say. ;)

Reply:
Do we really need to keep track of the whole education history of the person ? Wouldn't just be current one be enough ?

Reply:
Well, we want to know if it's their first year at school or not (for elections, secret…)

Reply:
In that case you'll only need to know if they are currently in 1A, 2A..., not what were their history

Reply:
Imagine someone who repeated their 1A.
Or an AST, who directly goes to 2A.
In both cases you won't know

Reply:
Let's keep 
- `cursus_type`: "IC", "EV"...
- `year`: 1A, 2A, gap year, 3A
- `year_of_entry`: 2017, ...